gtsam/hybrid/HybridConditional.h

```diff
@@ -162,7 +163,8 @@ class GTSAM_EXPORT HybridConditional
    * @return GaussianConditional::shared_ptr otherwise
    */
   GaussianConditional::shared_ptr asGaussian() const {
-    return std::dynamic_pointer_cast<GaussianConditional>(inner_);
+    if (!isContinuous()) return nullptr;
```

Comment:
Preferably throw exception instead of returning a nullptr since this can be a silent error/failure.

Reply:
Nope